compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs

```diff
@@ -768,15 +768,15 @@ typeOf' = \case
     checkExpr val ty2
     pure ty1
   EToInterface iface tpl val -> do
-    checkUniqueInterfaceInstanceExists (InterfaceInstanceHead iface tpl)
+    checkUniqueInterfaceInstanceExists (InterfaceInstanceHead iface tpl Nothing)
```

Comment:
having to add `Nothing`s here makes me think that the location field doesn't belong in `InterfaceInstanceHead`. Since the value of the new field is only used from `{T,I}PInterfaceInstance` and `pPrintInterfaceInstance`, I'd instead make the location a field of the former and an argument of the latter.

Reply:
on that note, could you please also handle the location in `pPrintInterfaceInstance`? it should be a matter of wrapping the existing body with `withSourceLoc lvl <maybe loc>`

Reply:
Yeah thats a fair point, I was treating the InterfaceInstanceHead as a more complete view of the instance than it really is, I've moved the source locations to the template/interface parts